test/functional/ui/cmdline_spec.lua

```diff
@@ -733,6 +735,74 @@ local function test_cmdline(linegrid)
 
 end
 
+describe('cmdline', function()
+  before_each(function()
+    clear()
+    screen = Screen.new(25, 5)
+    screen:attach({rgb=true})
+    screen:set_default_attr_ids({
+      [1] = {bold = true, foreground = Screen.colors.Blue1},
+      [2] = {reverse = true},
+      [3] = {bold = true, reverse = true},
+      [4] = {foreground = Screen.colors.Grey100, background = Screen.colors.Red},
+      [5] = {bold = true, foreground = Screen.colors.SeaGreen4},
+    })
+  end)
+
+  after_each(function()
+    screen:detach()
+  end)
+
+  describe('with timer', function()
+    it('draws correctly', function()
+      feed(':012345678901234567890123456789')
+
+      screen:expect{grid=[[
+                               |
+      {1:~                        }|
+      {3:                         }|
+      :012345678901234567890123|
+      456789^                   |
+      ]]}
+
+      command('call timer_start(0, {-> 1})')
```

Comment:
`run()` is not added here as adding it will cause the test to pass with or without the fix.

Without the fix, this test fails with:
```
Expected:
  |*                         |
  |{3:                         }|
  |:012345678901234567890123|
  |456789^                   |
  |456789^                   |
Actual:
  |*:012345678901234567890123|
  |:012345678901234567890123|
  |:012345678901234567890123|
  |:012345678901234567890123|
  |456789^                   |
```
The cmdline is redrawn multiple times, indicating that the test or Screen API is causing more than 1 `redrawcmdline`.
Adding `run()` seems to clear the extra cmdlines and results in the test not failing.

Reply:
What would be be the purpose of adding `run()`? Bare `run()` shouldn't be used in a screen test as it discards redraw events. Use `timeout` param to `screen:expect{}` if the purpose is to wait longer.